Sources/ArgumentParser/Completions/CompletionsGenerator.swift

```diff
@@ -164,11 +172,39 @@ extension ParsableCommand {
   }
 }
 
+extension [ParsableCommand.Type] {
+  // Include default 'help' subcommand in nonempty subcommand list iff no help subcommand already exists.
+  mutating func addHelpSubcommandIffMissing() {
+    if !isEmpty && allSatisfy({ $0._commandName != "help" }) {
+      append(HelpCommand.self)
+    }
+  }
+}
+
 extension Sequence where Element == ParsableCommand.Type {
```

Comment:
These can probably all be in the array extension…

Reply:
Do you want me to move all into:

```swift
extension [ParsableCommand.Type]
```

Or:

```swift
extension Sequence where Element == ParsableCommand.Type
```

The following also built on Swift 5.7.1 & passed all tests (I'm stuck on Swift 5.7.1 & Intel until the M4 Mac Studios are released…); I assume that the prior code is an older syntax, and that the following code is a newer equivalent syntax, but please let me know if I'm wrong (if correct, should a codebase-wide PR be done to migrate to the cleaner syntax?):

```swift
extension Sequence<ParsableCommand.Type>
```

The first is most succinct (& presumably handles all current use cases), but I assume the latter 2 are broader.

I'd probably vote for the third version, as long as it is correct…

Reply:
Re my original suggestion, I was just noticing that this was probably an unnecessarily broad extension, as I think these are all just used on arrays. It doesn't seem necessary to have the similar functionality broken across different extensions, so if it still builds with them all together, then that's a better strategy.

Those two syntaxes are equivalent, but the second has both Swift and runtime version requirements. The feature to support it (primary associated types) was added via [this proposal in Swift 5.7](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0358-primary-associated-types-in-stdlib.md), so we can use it without conditions now. (And a separate PR for that conversion would be great!)

Reply:
I actually need `completionFunctionName()` in the `Sequence` extension for changes that I've made to generated bash completion scripts in commits that I haven't yet submitted, so I will leave the two members where they are.